addons/mail/static/tests/qunit_suite_tests/models/file_uploader_tests.js

```diff
@@ -50,6 +53,70 @@ QUnit.test("no conflicts between file uploaders", async function () {
     await contains(".o_Chatter .o_AttachmentCard");
     await contains(".o_ChatWindow .o_Message .o_AttachmentCard");
 });
+
+QUnit.test('Arrows on nonviewable attachment', async function (assert) {
+    const pyEnv = await startServer();
+    const resPartnerId = pyEnv['res.partner'].create({});
+    const irAttachmentId = pyEnv['ir.attachment'].create({
+        mimetype: 'text/plain',
+        res_id: resPartnerId,
+        res_model: 'res.partner',
+    });
+    pyEnv['mail.message'].create({
+        attachment_ids: [irAttachmentId],
+        model: 'res.partner',
+        res_id: resPartnerId,
+    });
+    pyEnv['res.partner'].write([resPartnerId], {message_main_attachment_id : irAttachmentId})
+    const views = {
+        'res.partner,false,form':
+            '<form string="Partners">' +
+                '<sheet>' +
+                    '<field name="name"/>' +
+                '</sheet>' +
+                '<div class="o_attachment_preview"/>' +
+                '<div class="oe_chatter">' +
+                    '<field name="message_ids"/>' +
+                '</div>' +
+            '</form>',
+    };
+    patchUiSize({ size: SIZES.XXL });
+    const { messaging, openFormView } = await start({
+        mockRPC(route, args) {
+            if (_.str.contains(route, '/web/static/lib/pdfjs/web/viewer.html')) {
+                var canvas = document.createElement('canvas');
+                return canvas.toDataURL();
+            }
+        },
+        serverData: { views },
+    });
+    await openFormView({
+        res_id: resPartnerId,
+        res_model: 'res.partner',
+    });
+
+    // No viewer as text file is non viewable
+    await contains(".oe_chatter");
+    await contains(".o_attachment_preview", { count: 0 });
+    await contains(".o_attachment_preview_container > iframe", { count: 0 });
+    await contains(".arrow", { count: 0 });
+
+    // Add a PDF file
+    await click(".o_ChatterTopbar_buttonSendMessage");
+    await contains(".o_ComposerTextInput_textarea");
+    inputFiles(messaging.models["Chatter"].all()[0].composerView.fileUploader.fileInput, [
+        await createInputFile({ name: "invoice.pdf", contentType: "application/pdf" }),
+    ]);
```

Comment:
`contains("o_ComposerTextInput_textarea")` is not useful: this is already asserted a lot in many tests.

I assume it's to ensure good timing for inputing file? I would use `dragenterFiles` / `dropFiles` helpers already imported. Reason: they accept plain selectors that are visually visible on UI, e.g. `.o_Chatter_dropZone`. This means you don't have to find the `fileInput` by hand like you do right now.

(This is especially important for easing forward-port of this PR, as you may know 17.0 has discuss code completely rewritten and this becomes very different... Unless you use `dropFiles(".o_Chatter_dropZone")` which simply becomes `dropFiles(".o-mail-Dropzone")`)